velox/exec/MergeJoin.cpp

```diff
@@ -689,7 +774,14 @@ RowVectorPtr MergeJoin::getOutput() {
           if (isFullJoin(joinType_)) {
             rightIndex_ = 0;
           } else {
-            rightIndex_ = firstNonNull(rightInput_, rightKeys_);
+            auto firstNonNullIndex = firstNonNull(rightInput_, rightKeys_);
+            if (isRightJoin(joinType_) && firstNonNullIndex > 0) {
+              prepareOutput(nullptr, rightInput_);
+              for (auto i = 0; i < firstNonNullIndex; ++i) {
+                addOutputRowForRightJoin(rightInput_, i);
```

Comment:
what if `firstNonNullIndex` > `outputBatchSize_` ?

Reply:
@JkSelf

Reply:
@fzhedu The `firstNonNullIndex `will be equal to, but not greater than, `outputBatchSize_`.